### DISCUSSION TITLE
Successfully Enabled Jellyfin (via Casa Store) to use AMD GPU

### DISCUSSION THREAD

**Discussion by Dvalin21** [IceWhaleTech/CasaOS#801] [General]
Three things you need to get this going.

1. You need to edit the container and add the following
    a. Environment  Variable

   (Key)      DOCKER_MODS     (Value)    linuxserver/mods:jellyfin-amd    (Leave out Key and Value)
  
    b. Device
              
        (Host)     /dev/dri         (Container)  /dev/dri  (Leave out Host and Container)

2. Go into Jellyfin Dashboard/playback and change the HW to Vaapi and save.

3. You can install (My host is ubuntu server) [radeontop](https://github.com/clbr/radeontop) to verify that the gpu is betting used. But also you need to make sure that the libva-mesa-drivers are installed. (mesa seem to have been installed by default when I installed ubuntu server)

So far I have Frigate NVR and Jellyfin utilizing my AMD RX 5500M GPU.

Note: I found it better that the privilege option was ticked. Not sure why it wasn't default

**Comment by tigerinus** [comment]
This is a great post to be added to https://wiki.casaos.io/en/guides

**Reply by Dvalin21** [reply]
So after some more digging, there is an option in Jellyfin called AMD AMF which is strictly for AMD. It seems that only way to utilize this is install the AMDGPU Pro drivers sudo amdgpu-install --usecase=workstation --vulkan=pro -y --accept-eula
You will need to download the drivers from AMD first. However, I seem to be having an issue as the last kernel update Im not able to install these drivers. I did however posted at the AMD forum to see if they are already producing a fix. Main goal, I would like to be able to use AMD GPU as Tensorflow Detector with Frigate NVR. AMD has released the software needed, but suck when it comes to keeping it working on linux. OH!!! And by the way I removed vaapi-mesa and installed the amdgpu and these drivers seem to allow for greater usage than the vaapi-mesa. Here is what I used to install the amdgpus

sudo amdgpu-install --usecase=rocm,lrt,mllib,opencl,dkms --vulkan=pro -y --accept-eula

**Comment by cambiph** [comment]
How did you provide power to your GPU? I read in many places that using an standard ATX PSU is a bad idea since the motherboard cannot manage the power flow to the card an cause molten cables etc.

**Reply by Dvalin21** [reply]
I didn't do anything different. Matter of fact I just installed two gpus on the same psu. A AMD 5500XT and Nvidia 1660 Super on a 730W PSU. Running fine. Also I Have a lot fans.
